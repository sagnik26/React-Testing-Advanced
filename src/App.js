import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const [data, setData] = useState("");
  const [value, setValue] = useState("")

  return (
    <div className="App">
      <p>First React Test caseee</p>
      <input 
        type='text' 
        placeholder='Enter User Name' 
        id='userId' 
        name='username'
        value={data}
        onChange={(e) => setData(e.target.value)}
      />
      <br/>
      <br/>
      <br/>
      <h1>Test Click Event with Button</h1>
      <button onClick={() => setValue("Updated Data")}>Update data</button>
      <h2>{value}</h2>
      <img title='React logo' src={logo} />
    </div>
  );
}

export default App;
