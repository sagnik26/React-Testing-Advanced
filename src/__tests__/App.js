import { fireEvent, render, screen } from '@testing-library/react';
import App from '../App';

describe("Snapshot Testing", () => {
  test("App component snapshot testing", () => {
    const container = render(<App />);
    expect(container).toMatchSnapshot();
  })
})

describe("UI Testcases", () => {
  test("React First Test Case", () => {
    render(<App />);
    const text = screen.getByText(/First React Test Case/i);
    const title = screen.getByTitle("React logo");
    expect(text).toBeInTheDocument();
    expect(title).toBeInTheDocument();
  });
  
  test("testing Input Box", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    let checkInputPlaceHolder = screen.getByPlaceholderText("Enter User Name");
    expect(checkInput).toBeInTheDocument();
    expect(checkInputPlaceHolder).toBeInTheDocument();
    expect(checkInput).toHaveAttribute("name", "username");
    expect(checkInput).toHaveAttribute("id", "userId");
    expect(checkInput).toHaveAttribute("type", "text");
  });
})

describe("API Testcases", () => {
  test("API Test Case 1", () => {
    render(<App />);
    const text = screen.getByText(/First React Test Case/i);
    const title = screen.getByTitle("React logo");
    expect(text).toBeInTheDocument();
    expect(title).toBeInTheDocument();
  });
  
  test("API Test Case 2", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    let checkInputPlaceHolder = screen.getByPlaceholderText("Enter User Name");
    expect(checkInput).toBeInTheDocument();
    expect(checkInputPlaceHolder).toBeInTheDocument();
    expect(checkInput).toHaveAttribute("name", "username");
    expect(checkInput).toHaveAttribute("id", "userId");
    expect(checkInput).toHaveAttribute("type", "text");
  });

  describe("inner describe test group", () => {
    test("inner testcases", () => {
      render(<App />);
      let checkInput = screen.getByRole("textbox");
      expect(checkInput).toBeInTheDocument();
    })
  })
})

// Testing for onChange event
describe("inputbox testing", () => {
  it("input box onchange event test", () => {
    render(<App/>)
    let input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: "a" } });
    expect(input.value).toBe("a");
  })
})

// testing for onClick event
describe("Button Click Testing", () => {
  test("click event test cases", () => {
    render(<App/>);
    const btn = screen.getByRole("button");
    fireEvent.click(btn);
    expect(screen.getByText("Updated Data")).toBeInTheDocument();
  })
})


