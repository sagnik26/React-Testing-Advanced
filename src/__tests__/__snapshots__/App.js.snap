// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing App component snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          First React Test caseee
        </p>
        <input
          id="userId"
          name="username"
          placeholder="Enter User Name"
          type="text"
          value=""
        />
        <br />
        <br />
        <br />
        <h1>
          Test Click Event with Button
        </h1>
        <button>
          Update data
        </button>
        <h2 />
        <img
          src="logo.svg"
          title="React logo"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        First React Test caseee
      </p>
      <input
        id="userId"
        name="username"
        placeholder="Enter User Name"
        type="text"
        value=""
      />
      <br />
      <br />
      <br />
      <h1>
        Test Click Event with Button
      </h1>
      <button>
        Update data
      </button>
      <h2 />
      <img
        src="logo.svg"
        title="React logo"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
